solc/CommandLineInterface.cpp

```diff
@@ -622,14 +623,14 @@ bool CommandLineInterface::readInputFilesAndConfigureRemappings()
 				}
 
 				// NOTE: we ignore the FileNotFound exception as we manually check above
-				m_sourceCodes[infile.generic_string()] = readFileAsString(infile.string());
+				m_fileReader.setSource(infile.generic_string(), infile, readFileAsString(infile.string()));
```

Comment:
I think it would be better to pass in only `infile` and let the reader do other transformations, like calling `generic_string()`.